crates/solver-service/src/apis/quote/cost/engine.rs

```diff
@@ -0,0 +1,375 @@
+use alloy_primitives::U256;
+use solver_config::Config;
+use solver_core::SolverEngine;
+use solver_types::{
+	Address, ExecutionParams, FillProof, Order, OrderStatus, Transaction, TransactionHash,
+};
+use solver_types::{CostComponent, Quote, QuoteCost, QuoteError, QuoteOrder, SignatureType};
+
+#[derive(Debug, Clone)]
+struct PricingConfig {
+	currency: String,
+	commission_bps: u32,
+	gas_buffer_bps: u32,
+	rate_buffer_bps: u32,
+	enable_live_gas_estimate: bool,
+}
+
+impl PricingConfig {
+	fn from_config(config: &Config) -> Self {
```

Comment:
Probably we should pass the relevant params needed for this implementation as part of constructor instead of passing entire config and parsing here..

Reply:
Maybe not every order strategy implementation is forced to configure these fields. So it would break for other implementation..